contracts/TokenHandler.sol

```diff
@@ -162,16 +137,10 @@ contract TokenHandler is ITokenHandler, ITokenManagerType, ReentrancyGuard, Crea
      */
     // slither-disable-next-line locked-ether
     function postTokenManagerDeploy(uint256 tokenManagerType, address tokenManager) external payable {
+        // For lock/unlock token managers, the ITS contract needs an approval from the token manager to transfer tokens on its behalf
         if (tokenManagerType == uint256(TokenManagerType.LOCK_UNLOCK) || tokenManagerType == uint256(TokenManagerType.LOCK_UNLOCK_FEE)) {
             ITokenManager(tokenManager).approveService();
         }
-
-        // Approve the gateway here. One-time infinite approval works for gateway wrapped tokens, and for most origin tokens.
-        // Approval can be refreshed in the future if needed for certain tokens via an upgrade, but realistically should never be exhausted.
-        if (tokenManagerType == uint256(TokenManagerType.GATEWAY)) {
-            address token = ITokenManager(tokenManager).tokenAddress();
-            _approveGateway(token, UINT256_MAX);
-        }
     }
 
     function _transferTokenFrom(address tokenAddress, address from, address to, uint256 amount) internal {
```

Comment:
Upon checking code coverage, the `_transferToken` function is no longer being used.
```
function _transferToken(address tokenAddress, address to, uint256 amount) internal {
    // slither-disable-next-line arbitrary-send-erc20
    IERC20(tokenAddress).safeTransfer(to, amount);
}
```